.github/workflows/_build_test_upload.yml

```diff
@@ -62,22 +62,46 @@ jobs:
           - macos-latest
           - ubuntu-latest
           - windows-latest
+          - macos-m1-12
         python-version:
           - 3.7
           - 3.8
           - 3.9
           - "3.10"
+        # Python 3.7 is not officially provided on Apple Silicon
+        exclude:
+          - os: macos-m1-12
+            python-version: 3.7
     steps:
-      - name: Setup Python ${{ matrix.python-version }}
-        if: ${{ ! startsWith( matrix.os, 'ubuntu' ) }}
```

Comment:
Previously, we excluded `ubuntu` from this step because `ubuntu` used docker. Now, since `macos` would use conda to install python, `windows` becomes the only platform that would use this step.
cc: @NivekT